exonum/src/node/basic.rs

```diff
@@ -195,16 +195,15 @@ impl NodeHandler {
     pub fn handle_peer_exchange_timeout(&mut self) {
         if !self.state.peers().is_empty() {
             let to = self.state.peers().len();
-            let gen_peer_id = || -> usize {
+            let peer_id = {
```

Comment:
Just a little nitpick.
I suppose a goal of this lambda is to avoid a calculation of `peer_id` if `self.state.peers().len() == 0`. In this case I guess we always have at least a one peer and this doesn't matter. But in general it's a good practice to use a lambda or a function reference in such cases.
P.S. By the way. What does say a clippy on this ?

Reply:
ooh, now I see the point...